inference-engine/src/inference_engine/compilation_context.hpp

```diff
@@ -0,0 +1,146 @@
+// Copyright (C) 2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <sys/types.h>
+#include <sys/stat.h>
+#ifndef WIN32
+#include <unistd.h>
+#endif
+
+#include <string>
+#include <vector>
+#include <map>
+#include <xml_parse_utils.h>
+
+#include "ie_itt.hpp"
+#include "cpp_interfaces/exception2status.hpp"
+#include "transformations/serialize.hpp"
+#include "cnn_network_ngraph_impl.hpp"
+#include "cpp/ie_cnn_network.h"
+#include "details/ie_exception.hpp"
+
+#include "ngraph/pass/pass.hpp"
+#include "ngraph/variant.hpp"
+#include "ngraph/function.hpp"
+#include "ngraph/opsets/opset6.hpp"
+
+#ifdef WIN32
+#define stat _stat
+#endif
+
+namespace InferenceEngine {
+
+struct NetworkCompilationContext final {
+    static std::string calculateFileInfo(const std::string& filePath) {
```

Comment:
should not we make sure that `filePath` is absolute path? E.g. running from different CWD folders may lead to different files usage